### PR TITLE
Add more overview info on docs front page

### PR DESCRIPTION
**Pull request by jasongrout**
This is probably still a bit wordy, but it's nice to have some high-level context on the front page.

Also I deleted the developer_docs page which seemed largely empty.

### CODE REVIEW

**Review by ibdafna — APPROVED** [review]
LGTM

**Comment by jasongrout** [comment]
Awesome, thanks for the review. We can keep iterating to make the docs more approachable.

**Comment by jasongrout** [comment]
@meeseeksdev please backport to 7.x
